src/derivatives.jl

```diff
@@ -0,0 +1,162 @@
+function derivative(A::LinearInterpolation{<:AbstractVector{<:Number}}, t::Number)
+    idx = findfirst(x -> x >= t, A.t) - 1
+    idx == 0 ? idx += 1 : nothing
```

Comment:
This looks pretty strange.
```suggestion
    idx == 0 && ( idx += 1 )
```